src/DataModelRegistry.hpp

```diff
@@ -45,15 +54,40 @@ class NODE_EDITOR_PUBLIC DataModelRegistry
     {
       _registeredModels[name] = std::move(uniqueModel);
     }
+
+    if (TypeConverter && _registeredTypeConverters.count(name) == 0)
+    {
+      std::unique_ptr<NodeDataModel>& registeredModelRef = _registeredModels[name];
+
+      //Type converter node should have exactly one input and output ports, if thats not the case, we skip the registration.
+      //If the input and output type is the same, we also skip registration, because thats not a typecast node.
+      if (registeredModelRef->nPorts(PortType::In) != 1 || registeredModelRef->nPorts(PortType::Out) != 1 ||
+        registeredModelRef->dataType(PortType::In, 0).id == registeredModelRef->dataType(PortType::Out, 0).id)
+      {
+        return;
+      }
+
+      TypeConverterItemPtr converter = std::make_unique<TypeConverterItem>();
+      converter->Model = registeredModelRef->clone();
+      converter->SourceType = converter->Model->dataType(PortType::In, 0);
+      converter->DestinationType = converter->Model->dataType(PortType::Out, 0);
+
+      _registeredTypeConverters[name] = std::move(converter);
```

Comment:
Is it really necessary to know the name of a `TypeConverter`? IMHO, what essential and unique is the pair of types.

Shouldn't we register converters in the structure like:
```
    using ConvertingTypesPair = std::pair<input_type, output_type>;
    std::map<ConvertingTypesPair, TypeConverterItem>
```

That would guarantee that for given two convertible types there is just one Converter.
Besides, it would let the function `DataModelRegistry::getTypeConverter` to be much simpler (get rid of the loop).